webv2/table/utilities.go

```diff
@@ -156,12 +154,52 @@ func updateTypeOfInterleaveColumn(interleaveTableSchema []InterleaveTableSchema,
 		}
 		interleaveTableSchema[tableIndex].InterleaveColumnChanges[colIndex].Type = previousType
 		interleaveTableSchema[tableIndex].InterleaveColumnChanges[colIndex].UpdateType = updateType
+		interleaveTableSchema[tableIndex].InterleaveColumnChanges[colIndex].Size = previousSize
+		interleaveTableSchema[tableIndex].InterleaveColumnChanges[colIndex].UpdateSize = newSize
 	}
 
 	return interleaveTableSchema
 }
 
-func createInterleaveColumnType(interleaveTableSchema []InterleaveTableSchema, tableIndex int, colIndex int, columnId string, colName string, previousType string, updateType string) []InterleaveTableSchema {
+func updateSizeOfInterleaveTableSchema(interleaveTableSchema []InterleaveTableSchema, table, columnId, colName, colType string, previousSize, newSize int) []InterleaveTableSchema {
+	tableIndex := isTablePresent(interleaveTableSchema, table)
+	interleaveTableSchema = createInterleaveTableSchema(interleaveTableSchema, table, tableIndex)
+	interleaveTableSchema = updateSizeOfInterleaveColumn(interleaveTableSchema, table, columnId, colName, colType, previousSize, newSize)
+	return interleaveTableSchema
+}
+
+func updateSizeOfInterleaveColumn(interleaveTableSchema []InterleaveTableSchema, table, columnId, colName, colType string, previousSize, newSize int) []InterleaveTableSchema {
+	tableIndex := isTablePresent(interleaveTableSchema, table)
+	colIndex := isColumnPresent(interleaveTableSchema[tableIndex].InterleaveColumnChanges, columnId)
+	interleaveTableSchema = createInterleaveColumnSize(interleaveTableSchema, tableIndex, colIndex, columnId, colName, colType, previousSize, newSize)
+	if tableIndex != -1 && colIndex != -1 {
+		interleaveTableSchema[tableIndex].InterleaveColumnChanges[colIndex].ColumnId = columnId
+		if interleaveTableSchema[tableIndex].InterleaveColumnChanges[colIndex].ColumnName == "" {
+			interleaveTableSchema[tableIndex].InterleaveColumnChanges[colIndex].ColumnName = colName
+		}
+		if interleaveTableSchema[tableIndex].InterleaveColumnChanges[colIndex].Type == "" {
+			interleaveTableSchema[tableIndex].InterleaveColumnChanges[colIndex].Type = colType
+		}
+		interleaveTableSchema[tableIndex].InterleaveColumnChanges[colIndex].Size = previousSize
+		interleaveTableSchema[tableIndex].InterleaveColumnChanges[colIndex].UpdateSize = newSize
+	}
+	return interleaveTableSchema
+}
+
+func createInterleaveColumnSize(interleaveTableSchema []InterleaveTableSchema, tableIndex int, colIndex int, columnId, colName, colType string, previousSize, newSize int) []InterleaveTableSchema {
```

Comment:
What is the use case of adding a new column to interleaving ?

Reply:
removed